grunt_tasks/plugin.js

```diff
@@ -187,6 +184,18 @@ module.exports = function (grunt) {
                 config = grunt.file.readYAML(yml);
             }
 
+            var doAutoConfig = (
+                typeof config.grunt !== "object" ||
+                config.grunt.autoconf === (void 0) ||
```

Comment:
Your experience in javascript clear predates mine.  I've never heard of such a thing.